[SUPPORT] Set PGID and PUID in environment variables

**Describe your question or issue**
I'm trying to install MeshMonitor on the docker Docker on my Synology NAS but I have trouble starting the container because of permissions issues.
On other containers, when needed, I normally use PGID and PUID of the host user who own the data folder, but it seems to not work here.
Am I missing something ?

**Environment (please complete the following information):**
* Host OS: Synology DSM 7.1.1
* MeshMonitor Version: 2.21.13
* Deployment Method: Docker
* Docker/Container Version (if applicable): Latest version for DSM 7.1

**Configuration**
Please share relevant configuration files (remove any sensitive information):
* ~~docker-compose.yml (if using Docker Compose)~~
* Environment variables or .env file (redact sensitive values)
  * Default ones, I didn't changed them : 
    * PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    * NODE_VERSION=24.12.0
    * YARN_VERSION=1.22.22
    * NODE_ENV=production
    * PORT=3001
    * APPRISE_CONFIG_DIR=/data/apprise-config
    * APPRISE_STATEFUL_MODE=simple
  * Custom ones : 
    * MESHTASTIC_NODE_IP=[IP of my node]
    * PGID=[Group ID of the user who owns the "data" folder on the host]
    * PUID=[User ID  of the user who owns the "data" folder on the host]
* ~~Any custom configuration files~~

**Logs**
Please provide relevant log output:
```
Deploying scripts to /data/scripts/...
✓ Upgrade watchdog script deployed
✓ Docker socket test script deployed
2026-01-02 19:02:19,979 INFO Set uid to user 0 succeeded
2026-01-02 19:02:19,987 INFO supervisord started with pid 1
2026-01-02 19:02:20,990 INFO spawned: 'apprise' with pid 18
2026-01-02 19:02:20,994 INFO spawned: 'meshmonitor' with pid 19
2026-01-02 19:02:21,995 INFO success: apprise entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2026-01-02 19:02:21,996 INFO success: meshmonitor entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)

> meshmonitor@2.21.13 start
> node dist/server/server.js

[WARN] 
[WARN] ═══════════════════════════════════════════════════════════
[WARN] ⚠️  SESSION_SECRET NOT SET - USING AUTO-GENERATED SECRET
[WARN] ═══════════════════════════════════════════════════════════
[WARN]    For basic/home use, this is OK. Sessions will work.
[WARN]    
[WARN]    For production deployments with HTTPS, set SESSION_SECRET:
[WARN]    SESSION_SECRET=$(openssl rand -hex 32)
[WARN]    
[WARN]    ⚠️  Sessions will be reset on each container restart!
[WARN] ═══════════════════════════════════════════════════════════
[WARN] 
[WARN] ⚠️  COOKIE_SECURE not set - defaulting to false for HTTP compatibility
[WARN]    If using HTTPS, set COOKIE_SECURE=true for better security
[ERROR] ❌ DATABASE STARTUP ERROR ❌
[ERROR] ═══════════════════════════════════════════════════════════
[ERROR] Failed to access database directory or file
[ERROR] 
[ERROR] Database path: /data/meshmonitor.db
[ERROR] Database directory: /data
[ERROR] 
[ERROR] PERMISSION DENIED - The database directory or file is not writable.
[ERROR] 
[ERROR] For Docker deployments:
[ERROR]   1. Check that your volume mount exists and is writable
[ERROR]   2. Verify permissions on the host directory:
[ERROR]      chmod -R 755 /path/to/your/data/directory
[ERROR]   3. Example volume mount in docker-compose.yml:
[ERROR]      volumes:
[ERROR]        - ./meshmonitor-data:/data
[ERROR] 
[ERROR] For bare metal deployments:
[ERROR]   1. Ensure the data directory exists and is writable:
[ERROR]      mkdir -p /data
[ERROR]      chmod 755 /data
[ERROR] ═══════════════════════════════════════════════════════════
file:///app/dist/services/database.js:117
            throw new Error(`Database directory access check failed: ${err.message}`);
                  ^

Error: Database directory access check failed: EACCES: permission denied, access '/data'
    at new DatabaseService (file:///app/dist/services/database.js:117:19)
    at file:///app/dist/services/database.js:4064:16
    at ModuleJob.run (node:internal/modules/esm/module_job:413:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:660:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:101:5)

Node.js v24.12.0
npm notice
npm notice New minor version of npm available! 11.6.2 -> 11.7.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.7.0
npm notice To update run: npm install -g npm@11.7.0
npm notice
2026-01-02 19:02:24,624 WARN exited: meshmonitor (exit status 1; not expected)
```


**What have you tried?**
Nothing more than trying with the PGID & PUID environment variables.

**Additional context**
In the initial start, the deployment of the scripts seems to work (I have files in my folder), **before** the line "INFO Set uid to user 0 succeeded"

--- a/docker/docker-entrypoint.sh
+++ b/docker/docker-entrypoint.sh
@@ -5,35 +5,65 @@ set -e
 # If PUID and/or PGID environment variables are set, modify the node user/group
 # to match those IDs. This is useful for NAS systems like Synology where the
 # host directory ownership may differ from the default container user.
+#
+# Note: Alpine Linux doesn't have usermod/groupmod, so we use sed to modify
+# /etc/passwd and /etc/group directly. This is the standard approach for Alpine.
 
 PUID=${PUID:-1000}
 PGID=${PGID:-1000}
+
+# Validate PUID/PGID are numeric and in valid range (0-65534)
+validate_id() {
+    local id="$1"
+    local name="$2"
+    if ! echo "$id" | grep -qE '^[0-9]+$'; then
+        echo "ERROR: $name must be a numeric value, got: $id" >&2
+        exit 1
+    fi
+    if [ "$id" -lt 0 ] || [ "$id" -gt 65534 ]; then
+        echo "ERROR: $name must be between 0 and 65534, got: $id" >&2
+        exit 1
+    fi
+}
+
+validate_id "$PUID" "PUID"
+validate_id "$PGID" "PGID"
 
 # Get current node user/group IDs
 CURRENT_UID=$(id -u node)
 CURRENT_GID=$(id -g node)
 
-# Only modify if IDs differ from current
+# Track if we need to update the GID in passwd (only if PGID actually changed)
+NEW_GID="$CURRENT_GID"
+
+# Only modify group if GID differs from current
 if [ "$PGID" != "$CURRENT_GID" ]; then
     echo "Setting node group GID to $PGID..."
     # Delete existing group with target GID if it exists (and isn't node's group)
     EXISTING_GROUP=$(getent group "$PGID" 2>/dev/null | cut -d: -f1 || true)
     if [ -n "$EXISTING_GROUP" ] && [ "$EXISTING_GROUP" != "node" ]; then
+        echo "  Removing conflicting group: $EXISTING_GROUP"
         delgroup "$EXISTING_GROUP" 2>/dev/null || true
     fi
-    # Modify node group GID
+    # Modify node group GID in /etc/group
     sed -i "s/^node:x:$CURRENT_GID:/node:x:$PGID:/" /etc/group
+    NEW_GID="$PGID"
 fi
 
+# Only modify user if UID differs from current
 if [ "$PUID" != "$CURRENT_UID" ]; then
     echo "Setting node user UID to $PUID..."
     # Delete existing user with target UID if it exists (and isn't node)
     EXISTING_USER=$(getent passwd "$PUID" 2>/dev/null | cut -d: -f1 || true)
     if [ -n "$EXISTING_USER" ] && [ "$EXISTING_USER" != "node" ]; then
+        echo "  Removing conflicting user: $EXISTING_USER"
         deluser "$EXISTING_USER" 2>/dev/null || true
     fi
-    # Modify node user UID and GID
-    sed -i "s/^node:x:$CURRENT_UID:$CURRENT_GID:/node:x:$PUID:$PGID:/" /etc/passwd
+    # Modify node user UID and GID in /etc/passwd
+    sed -i "s/^node:x:$CURRENT_UID:$CURRENT_GID:/node:x:$PUID:$NEW_GID:/" /etc/passwd
+elif [ "$NEW_GID" != "$CURRENT_GID" ]; then
+    # UID unchanged but GID changed - update GID reference in passwd
+    sed -i "s/^node:x:$CURRENT_UID:$CURRENT_GID:/node:x:$CURRENT_UID:$NEW_GID:/" /etc/passwd
 fi
 
 # Copy upgrade-related scripts to shared data volume

